script.js

```diff
@@ -1,69 +1,101 @@
-function computerPlay() {
-  const choice = ['rock','paper','scissors'];
-  return choice[Math.floor(Math.random()*choice.length)]
+const choices = ['rock', 'paper', 'scissors'];
+
+function randomInteger(max, min = 0) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
 }
 
+function getRandomChoice() {
+  return choices[randomInteger(choices.length) - 1];
+}
 
-function playOneRound(playerSelection, computerSelection) {
+function sanitize(input) {
+  return input.trim().toLowerCase()
+}
 
-  // invalid input
-  if(playerSelection === null || computerSelection === null) {
-    return "Invalid input, try again"
-  }
-  if(playerSelection !== "rock" && playerSelection !== "paper" && playerSelection !== "scissors" ) {
-    return "Invalid input, try again"
+function validate(input) {
+  const errors = [];
+
+  if (!input) {
+    errors.push('Input must not be empty');
   }
 
-  if(playerSelection.toLowerCase() === computerSelection.toLowerCase()) {
-    return "It's a tie!"
+  if (!choices.includes(input)) {
+    errors.push('Input must be one of the following: rock, paper or scissors');
   }
 
-  if (playerSelection.toLowerCase() === 'rock') {
-    if(computerSelection.toLowerCase() === 'scissors') { 
-      return `win! ${playerSelection} beats ${computerSelection}`;
-    }else {
-      return `lost! ${computerSelection} beats ${playerSelection}`;
-    }
+  if (errors.length) {
+    throw new Error(errors.join(', '));
   }
+}
 
-  if (playerSelection.toLowerCase() === 'paper') {
-    if(computerSelection.toLowerCase() === 'rock') { 
-      return `win! ${playerSelection} beats ${computerSelection}`;
-    }else {
-      return `lost! ${computerSelection} beats ${playerSelection}`;
-    }
+function getResult(player1, player2) {
+  const battleMap = {
+    rock: 'scissors',
+    paper: 'rock',
+    scissors: 'paper'
+  };
+
+  if (player1 === player2) {
+    return 'tie';
   }
 
-  if (playerSelection.toLowerCase() === 'scissors') {
-    if(computerSelection.toLowerCase() === 'paper') { 
-      return `win! ${playerSelection} beats ${computerSelection}`;
-    } else {
-      return `lost! ${computerSelection} beats ${playerSelection}`;
-    }
+  if (battleMap[player1] === player2) {
+    return 'win';
+  } else {
+    return 'lose';
   }
+}
 
+function getMessage(result, player1, player2) {
+  const messages = {
+    tie: () => `It's a tie!`,
+    win: (player1, player2) => `You win! ${player1} beats ${player2}`,
+    lose: (player1, player2) => `You lose! ${player2} beats ${player1}`
+  };
+
+  return messages[result](player1, player2);
 }
 
-function game() {  
+function playRound(input) {
+  const player1 = sanitize(input);
+  const player2 = getRandomChoice();
+  
+  validate(player1);
+
+  const result = getResult(player1, player2);
+  const message = getMessage(result, player1, player2);
 
-  let playerWins, computerWins, tie = 0;
+  return {
+    result,
+    message
+  };
+}
+
+function game() {
+  const scores = {
```

Comment:
I rather prefer storing related information in an object instead of using separate variables